rudder-core/src/main/scala/com/normation/cfclerk/xmlparsers/TechniqueParser.scala

```diff
@@ -312,13 +320,54 @@ class TechniqueParser(
    * @return A compatible variable corresponding to the entry xml
    */
   def parseCompatibleTag(xml: Node): Compatible = {
-    if(xml.label != COMPAT_TAG) throw new ParsingException("CompatibleParser was expecting a <%s> xml and get:\n%s".format(COMPAT_TAG, xml))
+    if(xml.label != COMPAT_TAG) throw new ParsingException("CompatibleParser was expecting a <%s> xml and got:\n%s".format(COMPAT_TAG, xml))
     val os = xml \ COMPAT_OS map (n =>
       OperatingSystem(n.text, (n \ "@version").text))
     val agents = xml \ COMPAT_AGENT map (n =>
       Agent(n.text, (n \ "@version").text))
     Compatible(os, agents)
   }
+
+  /* parse RUNHOOKS xml node, which look like that:
+    <RUNHOOKS>
+      <PRE name="package">
+        <PARAMETER name="package" value="vim"/>
+      </PRE>
+      <POST name="servive" if="debian">
+        <PARAMETER name="service" value="some value"/>
+        <PARAMETER name="a post command">/something/that/is/complicated "with" 'all sort of quote'</PARAMETER>
+      </POST>
+    </RUNHOOKS>
+  */
+  def parseRunHooks(id: TechniqueId, xml: Node): List[RunHook] = {
+    def parseOneHook(xml: Node, kind: RunHook.Kind): RunHook = {
+      def opt(s: String) = if(s == null || s == "") None else Some(s)
+
+      (xml \ "@name").text match {
+        case null | "" => throw new ParsingException(s"Error: in technique '${id.toString()}', tried to parse a run hooks, but attribute 'name' is missing in: ${xml}")
+        case name      =>
+          RunHook(name, kind, (xml \ "@condition").text, (xml \\ "PARAMETER").toList.flatMap(p =>
```

Comment:
i don't see the "if" parsed here, but rather a condition
so documentation is invalid, it should not be "if" but "condition"